src/pages/index.tsx

```diff
@@ -0,0 +1,193 @@
+/*
+  Creates a search button that call the github api
+  and returns the data of the searched user
+
+  The data is client side fetched
+*/
+import { useCallback, useRef, useState } from 'react'
+import type { NextPage } from 'next'
+import Head from 'next/head'
+import Image from 'next/image'
+import { useEffect } from 'react'
+import styles from '../styles/Home.module.css'
+import { type } from 'os'
+
+const server = 'https://api.github.com'
+const api_path = 'users'
+
+type GithubUser = {
+  login: string
+  id: number
+  node_id: string
+  avatar_url: string
+  gravatar_id: string
+  url: string
+  html_url: string
+  followers_url: string
+  following_url: string
+  gists_url: string
+  starred_url: string
+  subscriptions_url: string
+  organizations_url: string
+  repos_url: string
+  events_url: string
+  received_events_url: string
+  type: string
+  site_admin: boolean
+  name: string
+  company: string
+  blog: string
+  location: string
+  email: string
+  hireable: boolean
+  bio: string
+  twitter_username: string
+  public_repos: number
+  public_gists: number
+  followers: number
+  following: number
+  created_at: string
+  updated_at: string
+}
+
+type Error = {
+  message: string
+  documentation_url: string
+}
+
+const Home: NextPage = () => {
+  // States
+  let username_input: string = '';
+  const [is_data_valid, setIsDataValid] = useState<boolean>(false)
+  const [data, setData] = useState<GithubUser>({
+    login: '',
+    id: 0,
+    node_id: '',
+    avatar_url: '',
+    gravatar_id: '',
+    url: '',
+    html_url: '',
+    followers_url: '',
+    following_url: '',
+    gists_url: '',
+    starred_url: '',
+    subscriptions_url: '',
+    organizations_url: '',
+    repos_url: '',
+    events_url: '',
+    received_events_url: '',
+    type: '',
+    site_admin: false,
+    name: '',
+    company: '',
+    blog: '',
+    location: '',
+    email: '',
+    hireable: false,
+    bio: '',
+    twitter_username: '',
+    public_repos: 0,
+    public_gists: 0,
+    followers: 0,
+    following: 0,
+    created_at: '',
+    updated_at: ''
+  })
+
+  const onClickButton = async () => {
+    username_input = (
+      document.getElementById('username_input') as HTMLInputElement
+    ).value
+    if (username_input == '') {
+      setIsDataValid(false)
+      return
+    }
+
+    fetch(`${server}/${api_path}/${username_input}`)
```

Comment:
You are hitting the github api without an Authorization Token, you are going to be rate limited after a few requests.

Since the client can see this code in the browser, you can't expose the token here, so you need to use your backend api.

Client -> /your/endpoint -> /github/api/endpoint with authorization header

In the requirements there's a link explaining how to create a personal token to use on these requests.